src/components/form/adapters/widget-adapter.ts

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { WidgetProps } from '../widgets/types';
+import { LimeElementsAdapter } from './base-adapter';
+
+function capitalize(text: string = '') {
+    return text.charAt(0).toUpperCase() + text.slice(1);
+}
+
+interface WidgetAdapterProps {
+    name: string;
+    value?: any;
+    widgetProps: WidgetProps;
+    extraProps?: any;
+    events?: { [key: string]: (event: any) => void };
+}
+
+export class LimeElementsWidgetAdapter extends React.Component {
+    state = {
+        modified: false,
+    };
+
+    constructor(public props: WidgetAdapterProps) {
+        super(props);
+
+        this.handleBlur = this.handleBlur.bind(this);
+        this.initState();
+    }
+
+    private initState() {
+        if (this.hasValue()) {
+            this.state.modified = true;
+        }
```

Comment:
Is this the correct behavior? If the form is initialized with a value, should it be considered to be modified? :thinking:

Reply:
that was the existing behavior before i broke LimeElementsAdapter into LimeElementsAdapter and LimeElementsWidgetAdapter. I believe you wrote it.

Reply:
Haha that's very possible, but it does not mean it is correct :smile:

Reply:
i think the difference would be, on a blank empty form we doe not immediately show validation messages until the user blurs the element, but when loading a form with existing data, we set modified to true which means errors will be shown immediately. It sounds like good behavior to me